validate_sample.py

```diff
@@ -0,0 +1,340 @@
+        _check_length(column, region, count, maximum_length)
+
+
+def validate_htc_willingness(count, sample_row):
+    column = 'HTC_WILLINGNESS'
+    mandatory = True
+    if _check_column_exists(column, mandatory, sample_row):
+        htc_willingness = sample_row[column]
+        if htc_willingness.isnumeric and len(htc_willingness) == 1:
+            return
+        print(f'Line {count}: {column} {htc_willingness} is not valid.')
+
+
+def validate_htc_digital(count, sample_row):
+    column = 'HTC_DIGITAL'
+    mandatory = True
+    if _check_column_exists(column, mandatory, sample_row):
+        htc_digital = sample_row[column]
+        if htc_digital.isnumeric and len(htc_digital) == 1:
+            return
+        print(f'Line {count}: {column} {htc_digital} is not valid.')
+
+
+def validate_fieldcordinator_id(count, sample_row):
+    column = 'FIELDCOORDINATOR_ID'
+    maximum_length = 7
+    mandatory = True
+    if _check_column_exists(column, mandatory, sample_row):
+        fieldcordinator_id = sample_row[column]
+        _check_length(column, fieldcordinator_id, count, maximum_length)
+
+
+def validate_fieldofficer_id(count, sample_row):
+    column = 'FIELDOFFICER_ID'
+    maximum_length = 10
+    mandatory = True
+    if _check_column_exists(column, mandatory, sample_row):
+        fieldofficer_id = sample_row[column]
+        _check_length(column, fieldofficer_id, count, maximum_length)
+
+
+def validate_treatment_code(count, sample_row):
+    column = 'TREATMENT_CODE'
+    maximum_length = 10
+    mandatory = True
+    if _check_column_exists(column, mandatory, sample_row):
+        treatment_code = sample_row[column]
+        _check_length(column, treatment_code, count, maximum_length)
+
+
+def validate_mmstreet_toid(count, sample_row):
+    column = 'MMSTREET_TOID'
+    maximum_length = 20
+    mandatory = False
+    if _check_column_exists(column, mandatory, sample_row):
+        mmstreet_toid = sample_row[column]
+        _check_length(column, mmstreet_toid, count, maximum_length)
+
+
+def validate_mmtopo_toid(count, sample_row):
+    column = 'MMTOPO_TOID'
+    maximum_length = 20
+    mandatory = False
+    if _check_column_exists(column, mandatory, sample_row):
+        mmtopo_toid = sample_row[column]
+        _check_length(column, mmtopo_toid, count, maximum_length)
+
+
+def validate_bng_northing(count, sample_row):
+    # not included in file to rm
+    pass
+
+
+def validate_bng_easting(count, sample_row):
+    # not included in file to rm
+    pass
+
+
+def validate_ce_expected_capacity(count, sample_row):
+    column = 'CE_EXPECTED_CAPACITY'
+    maximum_length = 4
+    mandatory = False
+    if _check_column_exists(column, mandatory, sample_row):
+        value = sample_row[column]
+        _check_length(column, value, count, maximum_length)
+        if not value.isnumeric():
+            print(f'Line {count}: {column}: {value} is not a valid integer.')
+
+
+def _check_column_exists(column, mandatory, sample_row):
```

Comment:
This check only finds completely missing columns. It does not catch if the column exists for a mandatory value, but the value in a row is empty.